manifest: Update sdk-zephyr to bring MPSL flash sync timeout fix

MarekPieta opened the pull request:
Change cherry-picks commit that allows to workaround MPSL flash sync timeout issue.

Jira: NCSDK-29354
Jira: DRGN-23363

NordicBuilder commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| zephyr | https://github.com/nrfconnect/sdk-zephyr/commit/d3d3f66845737630687c9018cdb687319b74ec17 | https://github.com/nrfconnect/sdk-zephyr/pull/2107 | [nrfconnect/sdk-zephyr#2107/files](https://github.com/nrfconnect/sdk-zephyr/pull/2107/files) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

NordicBuilder commented:
## CI Information

<sup>_To view the history of this post, clich the 'edited' button above_</sup>
Build number: 1
### Inputs:
#### Sources:
sdk-nrf: PR head: [86c2468cddb1608c92f8c4be46b1e83990b7d7f1](https://github.com/nrfconnect/sdk-nrf/pull/17885/commits/86c2468cddb1608c92f8c4be46b1e83990b7d7f1)
zephyr: PR head: [3ca920537c6ab6636b53ee3590bf1f5e8cd1e1e4](https://github.com/nrfconnect/sdk-zephyr/pull/2107/commits/3ca920537c6ab6636b53ee3590bf1f5e8cd1e1e4)
<details><summary> more details</summary>

sdk-nrf:
> PR head: [86c2468cddb1608c92f8c4be46b1e83990b7d7f1](https://github.com/nrfconnect/sdk-nrf/pull/17885/commits/86c2468cddb1608c92f8c4be46b1e83990b7d7f1)
> merge base: [8c57b70271845738bea7f0b4fbcb6716e670f10b](https://github.com/nrfconnect/sdk-nrf/commit/8c57b70271845738bea7f0b4fbcb6716e670f10b)
> target head (main): [dffa6721909cb6df7ffff8e17bd224a024d682bc](https://github.com/nrfconnect/sdk-nrf/commit/dffa6721909cb6df7ffff8e17bd224a024d682bc)
> [Diff](https://github.com/nrfconnect/sdk-nrf/compare/8c57b70271845738bea7f0b4fbcb6716e670f10b..86c2468cddb1608c92f8c4be46b1e83990b7d7f1)

zephyr:
> PR head: [3ca920537c6ab6636b53ee3590bf1f5e8cd1e1e4](https://github.com/nrfconnect/sdk-zephyr/pull/2107/commits/3ca920537c6ab6636b53ee3590bf1f5e8cd1e1e4)
> merge base: [d3d3f66845737630687c9018cdb687319b74ec17](https://github.com/nrfconnect/sdk-zephyr/commit/d3d3f66845737630687c9018cdb687319b74ec17)
> target head (main): [187d590eb8c82e4459700186b7437965191af4af](https://github.com/nrfconnect/sdk-zephyr/commit/187d590eb8c82e4459700186b7437965191af4af)
> [Diff](https://github.com/nrfconnect/sdk-zephyr/compare/d3d3f66845737630687c9018cdb687319b74ec17..3ca920537c6ab6636b53ee3590bf1f5e8cd1e1e4)
</details>

#### Github labels
| Enabled | Name | Description |
| -- | -- | -- |
|| ci-disabled | _Disable the ci execution_ | 
|| ci-all-test | _Run all of ci, no test spec filtering will be done_ | 
|| ci-force-downstream | _Force execution of downstream even if twister fails_ | 
|| ci-run-twister | _Force run twister_ | 
|| ci-run-zephyr-twister | _Force run zephyr twister_ | 

<details><summary> List of changed files detected by CI (24)</summary>

```
subsys
│  ├── bluetooth
│  │  ├── controller
│  │  │  │ Kconfig
west.yml
zephyr
│  ├── drivers
│  │  ├── misc
│  │  │  ├── coresight
│  │  │  │  │ Kconfig
│  ├── dts
│  │  ├── riscv
│  │  │  ├── nordic
│  │  │  │  │ nrf54h20_cpuflpr.dtsi
│  ├── samples
│  │  ├── boards
│  │  │  ├── nordic
│  │  │  │  ├── coresight_stm
│  │  │  │  │  ├── Kconfig.sysbuild
│  │  │  │  │  ├── boards
│  │  │  │  │  │  │ nrf54h20dk_nrf54h20_cpuapp.overlay
│  │  │  │  │  ├── remote
│  │  │  │  │  │  │ prj.conf
│  │  │  │  │  ├── sample.yaml
│  │  │  │  │  ├── src
│  │  │  │  │  │  │ main.c
│  │  │  │  │  │ sysbuild.cmake
│  ├── snippets
│  │  ├── nordic-log-stm-dict
│  │  │  ├── boards
│  │  │  │  │ nrf54h20_cpuapp.conf
│  │  │  │ snippet.yml
│  │  ├── nordic-log-stm
│  │  │  ├── boards
│  │  │  │  │ nrf54h20_cpuapp.conf
│  │  │  │ snippet.yml
│  ├── soc
│  │  ├── nordic
│  │  │  ├── nrf54h
│  │  │  │  ├── Kconfig.defconfig.nrf54h20_cpuflpr
│  │  │  │  │ Kconfig.defconfig.nrf54h20_cpuppr
│  │  │  ├── nrf54l
│  │  │  │  ├── Kconfig.defconfig.nrf54l15_cpuflpr
│  │  │  │  │ Kconfig.defconfig.nrf54l15_enga_cpuflpr
│  │  │  ├── nrf92
│  │  │  │  │ Kconfig.defconfig.nrf9280_cpuppr
│  ├── subsys
│  │  ├── bluetooth
│  │  │  ├── host
│  │  │  │  ├── Kconfig
│  │  │  │  ├── conn.c
│  │  │  │  ├── conn_internal.h
│  │  │  │  │ hci_core.c
│  │  ├── logging
│  │  │  ├── frontends
│  │  │  │  │ Kconfig
```
</details>

### Outputs:

#### Toolchain
Version: 
Build docker image: 

#### Test Spec & Results: :white_check_mark: Success; :x: Failure; :orange_circle: Queued; :yellow_circle: Progress; :white_medium_square: Skipped; :warning: Quarantine
- :orange_circle: Toolchain
- :orange_circle: Build twister
- :orange_circle: Integration tests
   - :orange_circle: test-sdk-audio
   - :orange_circle: desktop52_verification
   - :orange_circle: test_ble_nrf_config
   - :orange_circle: test-fw-nrfconnect-ble_samples
   - :orange_circle: test-fw-nrfconnect-chip
   - :orange_circle: test-fw-nrfconnect-nfc
   - :orange_circle: test-fw-nrfconnect-nrf-iot_cloud
   - :orange_circle: test-fw-nrfconnect-nrf-iot_thingy91
   - :orange_circle: test-fw-nrfconnect-rs
   - :orange_circle: test-fw-nrfconnect-fem
   - :orange_circle: test-fw-nrfconnect-thread
   - :orange_circle: test-fw-nrfconnect-zigbee
   - :orange_circle: test-sdk-find-my
   - :orange_circle: test-sdk-sidewalk
   - :orange_circle: test-low-level
   - :orange_circle: test-sdk-dfu

<details><summary>Disabled integration tests</summary>

- 
   - doc-internal
   - test-fw-nrfconnect-apps
   - test-fw-nrfconnect-ble_mesh
   - test-fw-nrfconnect-boot
   - test-fw-nrfconnect-nrf-iot_libmodem-nrf
   - test-fw-nrfconnect-nrf-iot_lwm2m
   - test-fw-nrfconnect-nrf-iot_mosh
   - test-fw-nrfconnect-nrf-iot_nrf_provisioning
   - test-fw-nrfconnect-nrf-iot_positioning
   - test-fw-nrfconnect-nrf-iot_samples
   - test-fw-nrfconnect-nrf-iot_serial_lte_modem
   - test-fw-nrfconnect-nrf-iot_zephyr_lwm2m
   - test-fw-nrfconnect-nrf_crypto
   - test-fw-nrfconnect-proprietary_esb
   - test-fw-nrfconnect-ps
   - test-fw-nrfconnect-rpc
   - test-fw-nrfconnect-tfm
   - test-sdk-mcuboot
   - test-sdk-pmic-samples
   - test-sdk-wifi
   - test-secdom-samples-public
</details>



*Note: This message is automatically posted and updated by the CI*

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-17885
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

rugeGerritsen commented on subsys/bluetooth/controller/Kconfig:
```diff
@@ -44,12 +44,17 @@ config BT_LL_SOFTDEVICE
 	select BT_CTLR_LE_PATH_LOSS_MONITORING_SUPPORT
 	select BT_CTLR_SUBRATING_SUPPORT
 	select BT_LL_SOFTDEVICE_HEADERS_INCLUDE
+	imply BT_CONN_TX_NOTIFY_WQ if SOC_FLASH_NRF_RADIO_SYNC_MPSL
```
I don't think it is a good idea to add an imply here for the SoftDevice Controller. We should rather enable it by default. That being said, we shouldn't enable experimental symbols for default builds

MarekPieta replied:
The symbol was marked as experimental, because it's a temporary solution and we want to eventually drop it without following deprecation policies. From my observations, enabling the option might improve stability and prevent flash operation failures.

If the solution is not accepted NCS-wide, we could enable it e.g. only for applications maintained by us. Still, looking at the possible consequences of the fixed flash issue, I think that it may be reasonable to enable it by default if MPSL flash sync is used together with Bluetooth.

MarekPieta replied:
To be honest I prefer `imply` over overriding Kconfig default here. While using `imply`, we do not need to worry about Kconfig dependencies. Overriding defaults could actually result in enabling a symbol even though dependencies of the symbol are not met (we would need to maintain the same set of dependencies in place where we override default to ensure consistent behavior).

MarekPieta replied:
Maybe we could introduce an artificial symbol that would be enabled by default if both SoftDevice and MPSL flash sync are enabled? The symbol could imply the workaround.

rugeGerritsen reviewed (changes requested):
I would suggest to split this PR into two parts: 
1. Bring in the changes from zephyr
2. Start using it

hermabe commented on subsys/bluetooth/controller/Kconfig:
```diff
@@ -44,12 +44,17 @@ config BT_LL_SOFTDEVICE
 
+	  The option implies using a separate workqueue for connection TX notify
+	  processing. This is done to prevent timeouts in MPSL flash sync when
+	  flash operations are done from the system workqueue context.
```
Are you doing blocking operations on the system workqueue? It is very much discouraged to block a shared resource.

rugeGerritsen replied:
We should look into making the flash API async

MarekPieta replied:
Actually, a lot of applications use system workqueue for flash operations (delegating settings store to system workq was quite common practice). Not sure if we can prevent users from doing it (we would need to e.g. add an execution context assertion to validate the requirement properly).

MarekPieta commented:
After discussion with codeowners, we decided not to enable the workaround by default (`CONFIG_BT_CONN_TX_NOTIFY_WQ` is experimental).
We will introduce a known issue with link to the `CONFIG_BT_CONN_TX_NOTIFY_WQ` as a workaround.

MarekPieta commented:
Dropped commit enabling the workaround by default and rebased onto main.
We will open separate PRs to enable the workaround in scope of our applications.

thst-nordic commented:
`CI/Jenkins/toolchain Failing after 5m — building toolchain`
rebase will fix this

MarekPieta commented:
> `CI/Jenkins/toolchain Failing after 5m — building toolchain` rebase will fix this

Rebased

rlubos commented:
Closing, will udpate SHA in other PR.
